client/src/partials/templates/findpatient.tmpl.html

```diff
@@ -1,73 +1,187 @@
+
+  <!-- "Panel" Find Patient Directive UI -->
+  <div class="panel panel-default"
+    ng-class="{'panel-success': session.loadStatus=='loaded', 'panel-danger': session.loadStatus=='error'}"
+    ng-switch-when="panel">
+
+    <!-- panel heading -->
+    <div class="panel-heading">
+      <i class="glyphicon glyphicon-search"></i> {{ "FIND.TITLE" | translate }}
+    </div>
+
+    <!-- panel body -->
+    <div class="panel-body">
 
-  <div class="panel-body find-collapse" ng-if="!(session.submitted && session.valid)">
-    <div ng-switch="session.state">
-
-      <!-- search by name -->
-      <div ng-switch-when="name">
-        <div class="input-group">
-          <input
-          type="text"
-          class="form-control"
-          ng-model="input.model"
-          uib-typeahead="patient as fmtPatient(patient) for patient in fuzzyNameSearch($viewValue)"
-          placeholder="{{ 'FIND.PATIENT_NAME' | translate }}"
-          uib-typeahead-wait-ms="150"
-          uib-typeahead-template-url="/partials/templates/debtorListItem.tmpl.html"
-          uib-typeahead-no-results="noPatients"
-          size="25">
-          <span class="input-group-btn">
-            <button ng-disabled="validateNameSearch(input.model)" ng-click="selectPatient(input.model)" class="btn btn-default">
-              {{ "FORM.SUBMIT" | translate }}
+      <div class="form-group has-feedback "
+        ng-class="{'has-success': session.loadStatus ==='loaded', 'has-error': session.loadStatus === 'error'}"
+        ng-form="formFindPatient"
+        novalidate>
+
+        <!-- input zone -->
+        <div class="input-group input-group-sm">
+
+          <!-- dropdown button -->
+          <!-- use of fixed width that correspond to notification text margin left -->
+          <div class="input-group-btn" uib-dropdown>
+            <button style="width:135px;" class="btn btn-default" uib-dropdown-toggle>
+              {{ session.selected.label | translate }} <span class="caret"></span>
             </button>
-          </span>
+            <ul class="uib-dropdown-menu">
+              <li ng-repeat="option in session.options">
+                <a href="" ng-click="findBy(option)">{{ option.label | translate }}</a>
+              </li>
+            </ul>
+          </div>
+          <!-- /dropdown button -->
+
+          <!-- search by ID  -->
+          <input ng-model="session.idInput"
+            ng-if="session.selected.key==0"
+            ng-keyup="$event.keyCode == 13 && submit()"
+            ng-change="validateNameSearch(session.idInput)"
+            name="idInput"
+            type="text"
+            class="form-control"
+            placeholder="{{ session.selected.placeholder | translate }}..."
+            required>
+
+          <!-- search by Name  -->
+          <input ng-model="session.nameInput"
+            ng-if="session.selected.key==1"
+            ng-keyup="$event.keyCode == 13 && submit()"
```

Comment:
This is kind of hacky.  Could you instead do something like this:
```html
<!-- search by name -->
<input ng-model="session.nameInput" ng-keyup="Ctrl.readInput($event)" >
```
```js
// inside your controller file
vm.readInput = readInput;

// determines if we need to submit by checking the keycode 
function readInput(event) {
  if (event.keyCode === 13) {
    submit();
  }
}
```
It is much clearer than using `&&` to conditionally execute logic.